sklearn/utils/tests/test_validation.py

```diff
@@ -539,3 +545,34 @@ def test_suppress_validation():
     assert_all_finite(X)
     sklearn.set_config(assume_finite=False)
     assert_raises(ValueError, assert_all_finite, X)
+
+
+class DummyMemory(object):
+    def __init__(self):
+        pass
+
+    def cache(self, func):
+        return func
+
+
+class WrongDummyMemory(object):
+    def __init__(self):
+        pass
+
+
+def test_check_memory():
+    memory = check_memory("TestString")
+    assert_equal(memory.cachedir, "TestString//joblib")
+    memory = check_memory(None)
+    assert_equal(memory.cachedir, None)
+    dummy = DummyMemory()
+    memory = check_memory(dummy)
+    assert memory is dummy
+    assert_raises_regex(ValueError, "'memory' is not a string "
+                        "or a Memory instance implementing a"
+                        " cache method. Got a {} instance, "
+                        "instead.".format(type(memory)), check_memory, 1)
+    assert_raises_regex(ValueError, "'memory' is not a string "
+                        "or a Memory instance implementing a cache method."
+                        " Got a {} instance, instead.".format(type(memory)),
```

Comment:
as previously mentioned, do not use type(memory). state directly string in your pattern